### PR TITLE
Use MongoDB's TTL index or capped collections for automatic cache culling

### PR DESCRIPTION
**Pull request by Olivier-OH**
Now, the cache backend never tries and cull older entries, and rather lets MongoDB do it itself.
For intensive usage, there may be a great performance gain, not doing this in the backend code.

### CODE REVIEW

**Comment by Alir3z4** [comment]
This is really good!
Thanks for the great work.

Do you have any further changes you need to apply ? otherwise I'll review and merge.

**Comment by Olivier-OH** [comment]
Thank you for the quick reply!

I think I'm good to merge. I've been using the code in a project here and it's been working fine.

**Comment by Olivier-OH** [comment]
Actually, there's something I've left out. I re-create a MongoClient instance each time I access the cache, which is absolutely ineficient. I'm going to correct this and push pretty soon.

**Comment by Olivier-OH** [comment]
After reviewing the code, I see that the MongoClient is created only if the collection is not created. The mongoclient is referenced by the collection, so everything is fine.

Sorry for the many comments :)

**Comment by Olivier-OH** [comment]
Woopsie, I just saw a bug I had left. On the set() method, the default value for timeout was `None`  instead of `DEFAULT_TIMEOUT` which led to cache entries having no expire date. This should happen only if an expire date of None or <0 is passed.

Corrected, pushed, and thus added to the merge request.

Cheers.

**Comment by Alir3z4** [comment]
@Olivier-OH thanks for taking care of the stuff and all the catches.
Would you be able to write some small tests for this ? I know this project doesn't have tests, but it would be super great if you could add some minimal tests, enough to just test the overall functionality.

Not a deal breaker though, if you're willing to and have time, we can merge and release this and make the tests in another pull requests, in that case we can use Travis-CI.

Django cache tests is a good place to look on how it's nicely done without writing much code: https://github.com/django/django/blob/master/tests/cache/tests.py

Anyway, we can merge this as it is, release and do that in another pull request.
Please let me know what do you think.

File: django_mongodb_cash_backend/__init__.py [hunk 1]
```diff
@@ -10,145 +11,222 @@
...
         else:
             return True
 
+    @reconnect()
     def get(self, key, default=None, version=None):
         coll = self._get_collection()
         key = self.make_key(key, version)
         self.validate_key(key)
-        now = datetime.utcnow()
+        now = timezone.now()
```
Comment (Alir3z4) [inline django_mongodb_cash_backend/__init__.py]: Excellent catch here.

**Comment by Olivier-OH** [comment]
I can write tests, but I'm not sure that I'll be able to do so before then end of August, though. For the rest, you may proceed at your convenience!

**Comment by Alir3z4** [comment]
Excellent, that's ok.
Thanks for the great work on this, this is absolutely amazing.
Keep up the great work.

I'm going to merge and will cut a release in the coming days.

**Pull request merged**